platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java

```diff
@@ -49,33 +49,30 @@ public DefaultAnchorShapeResolver(ConnectionWidget connection,
         this.attachedEnd = attachedTo;
         this.attachedWidget = attachedWidget;
     }
-    
+
     public int getEndLocation()
     {
-        int retVal = 0;
-        
+        int retVal;
+
         if(attachedWidget != null)
         {
             Rectangle bounds = attachedWidget.getBounds();
             retVal = getLocation(bounds);
         }
         else
         {
-//            Rectangle bounds = attachedWidget.getBounds();
-            Rectangle bounds = new Rectangle();
+            Widget sourceWidget;
             if(attachedEnd == ConnectionEnd.SOURCE)
             {
-                Widget sourceWidget = getConnection().getSourceAnchor().getRelatedWidget();
-                bounds = sourceWidget.getBounds();
+                sourceWidget = getConnection().getSourceAnchor().getRelatedWidget();
             }
             else
             {
-                Widget sourceWidget = getConnection().getTargetAnchor().getRelatedWidget();
-                bounds = sourceWidget.getBounds();
+                sourceWidget = getConnection().getTargetAnchor().getRelatedWidget();
             }
-            retVal = getLocation(bounds);
+            retVal = getLocation(sourceWidget.getBounds());
         }
-        
+
         return retVal;
```

Comment:
looking at the code again: this almost looks like a bug that it takes the `sourceWidget` bounds in both cases.

Lets not obfuscate this.

simply change
`Rectangle bounds = new Rectangle();`
into
`Rectangle bounds;`
if you want to get rid of the additional object and lets leave everything else as is.